s3/s3_test.go

```diff
@@ -0,0 +1,95 @@
+package s3
+
+import (
+	"flag"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/spf13/afero/test"
+)
+
+var s3bucket string
+
+func init() {
```

Comment:
Yeah, I can see where you're coming from.  Using S3 itself as an integration test sounds potentially slow (and $) too, but I guess that depends on a lot of factors.

With mock vs integration... neither way seems to be perfect, both being needed. :wink: